libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -461,8 +478,17 @@ void CodeTransform::operator()(FunctionDefinition const& _function)
 		}
 	}
 
-	m_assembly.appendLabel(m_context->functionExitPoints.top().label);
-	m_context->functionExitPoints.pop();
+	if (!subTransform.m_functionExitStackHeight.has_value())
+		subTransform.setupReturnVariablesAndFunctionExit();
+	appendPopUntil(*subTransform.m_functionExitStackHeight);
```

Comment:
Can we be certain that all retrun variables are below that height? Are there assertions for that?

Reply:
If there are return variables, then ``m_functionExitStackHeight`` is calculated as the maximum return variable slot plus one, so yes, they'll be below.